### ISSUE TITLE
DEBUG environment variable not available to tests (extension.test.ts)

### ISSUE THREAD

**Issue by stevenh** [microsoft/vscode#197494]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.83.1
- OS Version: Windows_NT x64 10.0.22621 (Windows 11)

Steps to Reproduce:

1. Set DEBUG environment variable and run an npm test for an extension.
2. Print the value of `process.env['DEBUG']` in the tests `extension.test.ts`

The output of `DEBUG` is blank.

Seems this was removed in response to https://github.com/microsoft/vscode/issues/130072 but I could find no mention of this anywhere, and had to search the code to find the issue, which I believe is caused by this: https://github.com/microsoft/vscode/blob/9aacac648821b5c9c1a7192ac9d44c50af6fe241/src/vs/base/common/processes.ts#L139

Every other variable is possible, but this is a problem because `DEBUG` is the variable that the [npm debug package](https://www.npmjs.com/package/debug) requires to control debugging.

**Comment by VSCodeTriageBot** [comment]
Thanks for creating this issue! It looks like you may be using an old version of VS Code, the latest stable release is 1.84.0. Please try upgrading to the latest version and checking whether this issue remains.

Happy Coding!

**Comment by stevenh** [comment]
No difference

**Comment by roblourens** [comment]
I don't think the line you're pointing at is actually relevant. That's only used for the remote vscode server for remote extensions (ssh, wsl, dev containers).

How are you setting the variable and running the test, can you be a little more specific?

**Comment by stevenh** [comment]
Running integration tests from the command line for example:
```shell
DEBUG=npm-groovy-lint npm run test
```

This runs the test but no debug is created by either the client or server componets that use the the npm debug library:
```javascript
const debug = require("debug")("vscode-groovy-lint");
```

**Comment by stevenh** [comment]
A test which demonstration this issue is available in [stevenh/vscode-testing](https://github.com/stevenh/vscode-testing).

**Comment by roblourens** [comment]
I see. I think your original code pointer is actually correct, I think I misread the references. And I think that we won't change this, but you can continue to work around it in the way you are. We don't want to allow passing environment variables through vscode processes when it may affect their behavior, because that will inevitably happen on accident and cause issues.

**Comment by stevenh** [comment]
Thanks for the feedback @roblourens.

Is there a way to improve this, for example being smarter about the clean up or at least clearly document this? I wasted a good amount of time investigating this before I found this seemingly undocumented behaviour and I'm sure it will effect others as `debug` is pretty much the standard for debug logging in the node ecosystem which is common for vscode extensions.

Thoughts?
